sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -268,7 +268,8 @@ public CheckpointMark getCheckpointMark() {
                         p.nextOffset,
                         p.lastWatermark.getMillis()))
             .collect(Collectors.toList()),
-        source.getSpec().isCommitOffsetsInFinalizeEnabled() ? Optional.of(this) : Optional.empty());
```

Comment:
This change seems to be no-op if previous logic worked. It changed from providing a Optional wrapped reader to set a boolean flag on same condition.

Would you mind sharing context why previously CheckpointMark did not have the reader available?

Reply:
We either need to add a condition for offsetBasedDeduplicationEnabled, or to always pass in. I thought optional was not necessary and it could always be passed, but okay with adding extra condition here instead.

Reply:
iiuc this PR is a preparation for future change, that need the reader for tasks other than CommitOffsetsInFinalize. Is that correct?

Reply:
I just reverted the change to remove optional to keep the changes simpler. Instead we pass in reader if offsetBasedDeduplicationSupported as well. This was a gap in a previously submitted pull request.